src/components/Form.js

```diff
@@ -0,0 +1,214 @@
+                if (!React.isValidElement(child)) {
+                    return child;
+                }
+
+                // Depth first traversal of the render tree as the input element is likely to be the last node
+                if (child.props.children) {
+                    return React.cloneElement(child, {
+                        children: childrenWrapperWithProps(child.props.children),
+                    });
+                }
+
+                // We check if the child has the isFormInput prop.
+                // We don't want to pass form props to non form components, e.g. View, Text, etc
+                if (!child.props.isFormInput) {
+                    return child;
+                }
+
+                // We clone the child passing down all form props
+                const inputID = child.props.inputID;
+                return React.cloneElement(child, {
+                    ref: node => this.inputRefs[inputID] = node,
+                    defaultValue: this.props.draftValues[inputID] || child.props.defaultValue,
+                    errorText: this.state.errors[inputID] || '',
+                    onBlur: (event) => {
+                        this.setTouchedInput(inputID);
+                        this.validate(this.getValues());
+                        if (child.props.onBlur) {
+                            child.props.onBlur(event);
+                        }
+                    },
+                    onChange: (value) => {
+                        // Expose value to the input ref so we can access it on native too
+                        this.inputRefs[inputID].value = value;
+
+                        if (child.props.shouldSaveDraft) {
+                            FormActions.setDraftValues(this.props.formID, {[inputID]: value});
+                        }
+                        if (this.touchedInputs[inputID]) {
+                            this.validate(this.getValues());
+                        }
+                    },
+                });
+            })
+        );
+
+        return (
+            <>
+                <ScrollView
+                    style={[styles.w100, styles.flex1]}
+                    contentContainerStyle={styles.flexGrow1}
+                    keyboardShouldPersistTaps="handled"
+                >
+                    <View style={[this.props.style]}>
+                        {childrenWrapperWithProps(this.props.children)}
+                        <FormAlertWithSubmitButton
+                            buttonText={this.props.submitButtonText}
+                            isAlertVisible={_.size(this.state.errors) > 0 || Boolean(this.props.formState.serverErrorMessage)}
+                            isLoading={this.props.formState.isSubmitting}
+                            message={this.props.formState.serverErrorMessage}
+                            onSubmit={this.submit}
+                            onFixTheErrorsLinkPressed={() => {
```

Comment:
Technically this arrow function isn't needed:

```suggestion
                            onFixTheErrorsLinkPressed={this.inputRefs[_.first(_.keys(this.state.errors))].focus}
```

Reply:
I think that would immediately try to read `focus` when the component is rendered, but the first ref might be undefined if we don't have errors and it would throw.

Reply:
Yup, it seems that we need to leave it here 😄 

![Screen Shot 2022-01-27 at 4 21 33 PM](https://user-images.githubusercontent.com/22219519/151459687-280d458b-e8c9-4da2-a6dd-7bfec24fde4a.png)

Reply:
NAB